src/Slices/InstanceComposerEditor/Core/helper.ts

```diff
@@ -0,0 +1,26 @@
+import { ServiceModel } from "@/Core";
+
+/**
+ * Recursively filters an array of ServiceModel objects based on possible ownership of the root instance.
+ *
+ * @param {ServiceModel[]} services - An array of ServiceModel objects to filter.
+ * @param {ServiceModel} mainService - root service to filter owned services by it.
+ * @returns {ServiceModel[]} An array of owned ServiceModel objects
+ */
+export const filterServices = (
+  services: ServiceModel[],
+  mainService: ServiceModel,
+): ServiceModel[] => {
+  const relatedServices = services.filter((service) =>
```

Comment:
I would personally be more defensive. Especially since this is a recursive method 
```

export const filterServices = (
  services: ServiceModel[] = [],
  mainService: ServiceModel | undefined,
): ServiceModel[] => {
  if (!mainService || !mainService.owned_entities || !Array.isArray(services)) {
    return [];
  } 
// rest of the code
```

Reply:
Added